source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,281 @@
+</blockquote>
+
+## Implement a GraphQL module
+
+A Front-Commerce GraphQL module has to export an object containing the different services it provides.
+
+In the previous GraphQL module registration example, the module definition file would be located at
+`src/server/modules/hello-world/index.js` with other module files in the same directory.
+
+Below are documented the available keys to define a Front-Commerce GraphQL module.
+
+### `namespace`
+
+The simplest (but useless) GraphQL module only requires an unique `namespace` key:
+
+```js
+// A minimal Front-Commerce GraphQL module that basically does nothing
+module.exports = {
+  namespace: "Acme/HelloWorld"
+};
+```
+
+This `namespace` can be used by other modules as a dependency, to ensure another module has been registered in the application.
+
+### `dependencies` (optional)
+
+A list of module namespaces which must have been registered in the application. Dependencies will be initialized before this module.
+
+Example:
+
+```js
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  dependencies: ["Acme/Core"],
+  // …
+};
+```
+
+<blockquote class="warning">
+**Note:** we recommend to try to avoid dependencies between your modules. Usually this is a smell that the schema may be implemented differently and modules are too coupled together. A good use case is for instance to depend on a Front-Commerce core module.
+</blockquote>
+
+<blockquote class="info">
+**In depth:** modules are sorted using the [toposort](https://www.npmjs.com/package/toposort) library. See `flattenAndReorderModulesUsingDependencies` code and tests to understand the implementation
+</blockquote>
+
+### `typeDefs` (optional)
+
+GraphQL type definitions provided by this module. As a developer, this is a contract with the rest of your codebase.
+
+Type definitions must [GraphQL schema language strings](https://graphql.org/learn/schema/), and can be declared inline or loaded from a `gql` file. One can create new types, add top level queries (by extending the `Query` type), mutations or extend types from other modules.
+
+```js
+// or import typeDefs from './schema.gql';
+const typeDefs = `
+extend type Query {
+  "Be polite!"
+  sayHello (name: String!): Message
+}
+
+type Message {
+  response: String
+  audio: String
+}
+`
+
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  typeDefs: typeDefs
+};
+```
+
+<blockquote class="info">
+**In depth:** under the hood, all module’s `typeDefs` are merged in a single array passed to graphql-tools' [`makeExecutableSchema`](https://www.apollographql.com/docs/graphql-tools/generate-schema.html#makeExecutableSchema) function
+</blockquote>
+
+### `resolvers` (optional)
+
+This is an object containing all the resolvers for resolving fields defined in the schema (usually from your `typeDefs`). This is where the implementation resides.
```

Comment:
Is it worth mentioning the types that are used as keys here (using Query as an example)? Even if it's documented in Apollo Tools I'd say it would make it easier for the readers not to switch context.